### PR TITLE
Add lateral operators fill_undef and disk_avg

### PR DESCRIPTION
**Pull request by cfkanesan**
## Purpose

Add lateral operators fill_undef and disk_avg.

## Code changes:

- Add module `lateral_operators` containing the `fill_undef` and `disk_avg` functions.

### CODE REVIEW

File: src/idpi/operators/lateral_operators.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+    undef = windows.isnull()
+    frac_undef = undef.where(mask).where(in_bnds_x).where(in_bnds_y).mean(weights.dims)
+    return frac_undef <= frac_val
+
+
+def fill_undef(field: xr.DataArray, radius: int, frac_val: float) -> xr.DataArray:
+    """Fill undefined values.
+
+    Replace undefined values with a value derived from neighbourhood.
```
Comment (cosunae) [inline src/idpi/operators/lateral_operators.py:116]: can you specify in the doc how the value from the neighbourhood is computed?

File: src/idpi/operators/lateral_operators.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+    Parameters
+    ----------
+    field : xarray.DataArray
+        Input field.
+    radius : int
+        Radius of the convolution window.
+    frac_val : float
+        Threshold of acceptable ratio of undefined values in window.
```
Comment (cosunae) [inline src/idpi/operators/lateral_operators.py:125]: it is unclear from the doc what happens if this threshold is surpassed

File: src/idpi/operators/lateral_operators.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+
+# Standard library
+from typing import Literal
+
+# Third-party
+import numpy as np
+import xarray as xr
+
+
+def compute_weights(
```
Comment (cosunae) [inline src/idpi/operators/lateral_operators.py:11]: perhaps we could unittest this function ? for some of the use cases and compare against a numpy array of the expected values

Comment (cosunae) [reply src/idpi/operators/lateral_operators.py:11]: I mean an isolated test, out of the regression that is using fieldextra

File: src/idpi/operators/lateral_operators.py [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+    Returns
+    -------
+    xarray.DataArray
+        Boolean mask indicating where the undefined value ratio in the convolution
+        kernel is within the acceptable threshold.
+
+    """
+    mask = weights > 0
+
+    loc_x = windows.x + windows.win_x - windows.sizes["win_x"] // 2
```
Comment (cosunae) [inline src/idpi/operators/lateral_operators.py:102]: this block is hard to follow, not sure if some comments would help to digest it better?

**Comment by cosunae** [comment]
I generally quite like it, I did run it to watch some of the weight patterns for the convolution, and seems all good to me. 
I just have some comments to improve mostly the documentation

File: src/idpi/operators/lateral_operators.py (hunk 4 above)
Comment (cfkanesan) [reply src/idpi/operators/lateral_operators.py:102]: For each window, `x` is the coordinate of the center of the window in the parent field. `win_x` is the local coordinate of the current element in the window. `loc_x` represents the location of the window element in the coordinates of the parent field. This is needed to filter out nans that are introducing by the padding of the field by the rolling method. Only the nans that are present within the bound of the field should count towards the `frac_val` threshold.

Comment (cfkanesan) [reply src/idpi/operators/lateral_operators.py:102]: Should I add this paragraph as a comment?

File: src/idpi/operators/lateral_operators.py (hunk 2 above)
Comment (cfkanesan) [reply src/idpi/operators/lateral_operators.py:125]: The value for that window remains undefined.

File: src/idpi/operators/lateral_operators.py (hunk 1 above)
Comment (cfkanesan) [reply src/idpi/operators/lateral_operators.py:116]: It's essentially a Gaussian filter with sigma = 1.

File: src/idpi/operators/lateral_operators.py (hunk 2 above)
Comment (cosunae) [reply src/idpi/operators/lateral_operators.py:134]: yes, I would add it to the comment

Comment (cfkanesan) [reply src/idpi/operators/lateral_operators.py:134]: I have added this in the latest commit

Comment (cosunae) [reply src/idpi/operators/lateral_operators.py:134]: sorry, I didnt realize

**Review by cosunae — APPROVED** [review]
